syntax = "proto2";

package detection.protos; 

import "detection/protos/optimizer.proto";


message TrainConfig {
  optional Optimizer optimizer = 1;

  optional int32 max_to_keep = 2 [default = 5];

  optional int32 num_steps = 3;

  optional int32 start_save_ckpt_after_n_steps = 4 [default = 0];

  optional int32 save_ckpt_every_n_steps = 5 [default = 10000];

  optional int32 print_progress_every_n_steps = 6 [default = 100]; 

  optional float gradient_clipping_by_norm = 7 [default = 0.0];

  repeated string input_file = 8;

  optional string load_ckpt_path = 9;

  optional string save_ckpt_path = 10;

  optional string checkpoint_type = 11 [default = 'classification'];

  optional bool include_global_step = 12 [default = true];
}
